### ISSUE TITLE
[BUG] Bed leveling fade effectively disabled, the default 10mm is overwritten with 0mm

### ISSUE THREAD

**Issue by CarlosGS** [MarlinFirmware/Marlin#17343]
The default fade height is [defined as 10mm](https://github.com/MarlinFirmware/Marlin/blob/9ade72aee94981b26be0ca125ab487166d0bd7e3/Marlin/src/module/settings.cpp#L669), which is a reasonable value.
**But it is [initialized as 0mm](https://github.com/MarlinFirmware/Marlin/blob/9ade72aee94981b26be0ca125ab487166d0bd7e3/Marlin/src/module/settings.cpp#L2550).**

So starting with an empty EEPROM, the fading is initially disabled. I don't think it's the desired behavior since it continuously updates Z during the whole print.

Possibly related:  #12410

(Tried to keep the bug report concise, thanks so much for this software, you are awesome!!)

**Comment by CarlosGS** [comment]
This is still an issue in the latest version.

**Comment by boelle** [comment]
Configurations, please
Please ZIP up your `Configuration.h` and `Configuration_adv.h` files (as
requested in the Issue template) and drop them into your next reply.
We'll check them over and see if anything is amiss.

**Comment by CarlosGS** [comment]
Hi, apart from ENABLE_LEVELING_FADE_HEIGHT, this bug is 100% configuration agnostic.

My bug report links to the problematic code lines (I've verified the two conflicting definitions are still present in the latest version).

The only thing left is to decide whether to keep 10mm or 0mm as the default. I'd suggest 10mm since 0mm keeps it enabled for the whole prints. Cheers! :)

**Comment by boelle** [comment]
so its more a config issue than a bug?

**Comment by CarlosGS** [comment]
I'd consider it an "initialization consistency" bug, since when configuring a new machine you can't enable level fade and expect an initialization of 10mm. At this moment there are two choices for users:
- On every fresh install (or eeprom clearance), manually setting a level height value, through the LCD or gcode commands, to overwrite the wrong 0.0mm.
- Manually modify configuration_store.cpp to guarantee a default initialization >0.0mm in any firmware updates.

**Otherwise, some users get 10mm default fading, and some 0mm -> whole print fading (!!)**. Sorry if I couldn't summarize it properly before: Changing `new_z_fade_height` 0.0 to 10.0 would solve the issue.

**Comment by CarlosGS** [comment]
Hi, as far as I can tell, the bug is still present:
- Fade defined as 10.0 in the configuration: https://github.com/MarlinFirmware/Marlin/blob/9ade72aee94981b26be0ca125ab487166d0bd7e3/Marlin/src/module/settings.cpp#L669
- Initialized as 0.0 when configuration is not available (i.e. first run): https://github.com/MarlinFirmware/Marlin/blob/9ade72aee94981b26be0ca125ab487166d0bd7e3/Marlin/src/module/settings.cpp#L2550
Cheers!

To summarize: The only thing left is to decide whether to keep 10mm or 0mm as the default. I'd suggest 10mm since 0mm keeps it enabled for the whole prints.

**Comment by qwewer0** [comment]
Maybe a setting in Configuration.h would be useful?

```
#define ENABLE_LEVELING_FADE_HEIGHT
#if ENABLED(ENABLE_LEVELING_FADE_HEIGHT)
  #define LEVELING_FADE_HEIGHT 10
#endif
```

**Comment by qwewer0** [comment]
After an `M502` my fade height too changes to **0**.

**Comment by qwewer0** [comment]
Is there a way to get an editable fade height in Configuration.h or Configuration_adv.h? So that value is kept after a default restore?

**Comment by CarlosGS** [comment]
+1 to that solution, it would be ideal.

(Latest version renamed the file, issue still present, I've updated the above references)

**Comment by qwewer0** [comment]
Still a problem.

**Comment by CarlosGS** [comment]
This is still present in the latest code.

To summarize: The only thing left is to decide whether to keep 10mm or 0mm as the default. I'd suggest 10mm since 0mm keeps it enabled for the whole prints.

**Comment by sjasonsmith** [comment]
@CarlosGS please review the pull request references above, and provide feedback. I believe this will satisfy your request if it is accepted.

**Comment by CarlosGS** [comment]
Yep! Brilliant! Thanks!! :D

**Comment by sjasonsmith** [comment]
This was merged, so I'll close this out.
